std/math.d

```diff
@@ -6235,12 +6235,16 @@ Unqual!(Largest!(F, G)) pow(F, G)(F x, G y) @nogc @trusted pure nothrow
         {
             // Result is real only if y is an integer
             // Check for a non-zero fractional part
-            if (y > -1.0 / real.epsilon && y < 1.0 / real.epsilon)
+            enum real maxPrecise = ulong.max;
+            enum real minPrecise = -maxPrecise;
+            if (y >= minPrecise && y <= maxPrecise)
             {
-                long w = cast(long)y;
-                if (w != y)
+                real absY = fabs(y);
+                ulong w = cast(ulong)absY;
+                if (w != absY)
```

Comment:
these three lines look strange. aren't you comparing y with y in the end. isn't the only logical reason for that been != floating point imprecision